src/sinks/util/tls.rs

```diff
@@ -55,32 +54,37 @@ enum TlsError {
 #[derive(Clone, Debug, Default, Deserialize, Serialize)]
 pub struct TlsOptions {
     pub verify_certificate: Option<bool>,
+    pub verify_hostname: Option<bool>,
     pub ca_path: Option<PathBuf>,
     pub crt_path: Option<PathBuf>,
     pub key_path: Option<PathBuf>,
     pub key_pass: Option<String>,
 }
 
-impl TlsOptions {
-    pub fn check_warnings(&self, sink: &str) {
-        if self.verify_certificate == Some(false) {
-            warn!(
-                "`verify_certificate` in {} sink is DISABLED, this may lead to security vulnerabilities", sink
-            );
-        }
-    }
-}
-
 /// Directly usable settings for TLS connectors
+#[derive(Clone, Default)]
 pub struct TlsSettings {
-    verify_certificate: bool,
+    accept_invalid_certificates: bool,
+    accept_invalid_hostnames: bool,
     authority: Option<Certificate>,
-    identity: Option<Identity>,
+    identity: Option<Vec<u8>>, // native_tls::Identity doesn't implement Clone yet
 }
 
-impl TryFrom<&TlsOptions> for TlsSettings {
-    type Error = crate::Error;
-    fn try_from(options: &TlsOptions) -> Result<Self, Self::Error> {
+impl TlsSettings {
+    pub fn from_options(options: &Option<TlsOptions>, sink: &str) -> crate::Result<Self> {
```

Comment:
If we remove the below we don't need to pass the sink in. Ideally, this config shouldn't be aware of sinks.

Reply:
I thought it would be important to indicate which sink the warning was associated with. If you don't think that is relevant, I would be happy to drop the extra parameter.

Reply:
It should be covered since this log should happen within a `tracing` span that indicates which sink this is coming from.